pkg/apicast/apicast.go

```diff
@@ -226,25 +217,26 @@ func (a *APIcast) Deployment() *appsv1.Deployment {
 					},
 				},
 			},
-			Replicas: &a.Replicas, // TODO set to nil?
+			Replicas: &a.options.Replicas,
 		},
 	}
-	if a.OwnerReference != nil {
-		addOwnerRefToObject(deployment, *a.OwnerReference)
+
+	if err := controllerutil.SetControllerReference(a.options.Owner, deployment, a.scheme); err != nil {
```

Comment:
On the original design we checked whether ownerReference was nil or not (and allowed it to be) because in the design of the APIcast class we wanted to allow defining an APIcast without an owner.
Do you think it's not worth it?

Reply:
I did not see any value making it optional. No event or condition leads to ownerless objects. If you think it is worth for any reason, let me know and I can make it optional.

Reply:
Ok. Lets leave it this way. We can always change it in the future again if it was needed, which does not seem so for the moment